stable/agent/templates/deployment.yaml

```diff
@@ -65,6 +65,10 @@ spec:
             {{- if .Values.dind.enabled }}
             - name: DOCKER_HOST
               value: "tcp://localhost:{{ .Values.dind.port | default "2375" }}"
+            - name: BUILDKITE_BUILD_PATH
```

Comment:
should we only set these if DinD is enabled? Kind of seems like any containerized setup could benefit from explicit settings.

Reply:
yes, I agree this should be not only for DinD

Reply:
Should [the shared volume](https://github.com/buildkite/charts/blob/master/stable/agent/templates/deployment.yaml#L92-L93) be created always as well?

Reply:
Based on https://kubernetes.io/docs/concepts/storage/volumes/#emptydir, seems like it should for sure in order to preserve buildkite data across pod restarts and crashes.

Reply:
@0x0I so is the `emptyDir` fine or should I create something more durable?

Reply:
a PVC would make sense here

Reply:
@rimusz I think that would require switching from a `Deployment` to a `StatefulSet`? I don't think you can mount a PVC multiple times in `ReadWrite` mode.

Reply:
yes, if we want to scale the agent, then yes StatefulSet would be better with it's own PVC per replica

Reply:
@rimusz 👍 Could we try that in another PR?

Reply:
:+1: agreed on the switch being implemented in another PR